src/components/banner.js

```diff
@@ -38,12 +46,7 @@ function Banner(props) {
     }, 10000);
     snippetChooser(0, snippets.length - 1);
     return () => clearInterval(interval);
-  }, [difference]);
-
-  const snippetChooser = (min, max) => {
-    const index = Math.random() * (max - min) + min;
-    setSnippet(snippets[Math.floor(index)]);
-  };
+  }, [difference, snippetChooser, snippets.length, start]);
```

Comment:
Is the "difference" really needed in the dep array for this effect ? Its value is not consumed in the effect.  
Also setDifference is being called which changes difference and results in effect being triggered again and so on so forth.

@jeremyphilemon  @ssonal

Reply:
Makes sense. Fixed, thanks!